pkg/hostagent/pods.go

```diff
@@ -432,13 +447,40 @@ func (agent *HostAgent) syncEps() bool {
 					continue
 				}
 				ep.ServiceClusterIps = agent.getServiceIPs(poduuid)
+
+				update_ts := time.Now().UTC()
+				agent.podNameToTimeStamps[poduuid].epFileUpdateTimestamps = append(agent.podNameToTimeStamps[poduuid].epFileUpdateTimestamps, update_ts)
```

Comment:
Do we need to acquire lock while accessing agent.podNameToTimeStamps?

Reply:
Yes @akhilamohanan, added the locks wherever necessary. Tested with race flag to check for any race condition. Thank you.